src/clj_deconstruct_sigs/core.cljc

```diff
@@ -0,0 +1,148 @@
+(ns clj-deconstruct-sigs.core)
+
+#?(:clj (set! *unchecked-math* :warn-on-boxed))
```

Comment:
I think it's duplicated to `:global-vars`. I don't know whether this form is necessary.

Reply:
I added the `:global-vars` in `:dev` profile to avoid unintentional boxing in tests and benchmarks while developing. It doesn't affect released jars and their dependents.
This line is for selectively enabling unchecked-math only for this namespace, regardless of the project settings.

Reply:
That makes sense. Thank you.